src/service/decorators.ts

```diff
@@ -73,14 +75,37 @@ export function decoratorValidator(decAst: CallExpression): AnyValidator | null
   const decoratorName = decAst.callee.name;
   if (decoratorName === Decorator.Enum) {
     const args = callParamsAstToSet(decAst);
-    return (expr: Expression): boolean =>
-      expr.type === Syntax.Literal && expr.value != null && args.has(expr.value.toString());
+    return (expr: Expression): DiagnosticError[] => {
+      if (expr.type !== Syntax.Literal) {
+        return [createError('Type must be Literal')];
+      }
+      if (expr.value === null) {
+        return [createError('Type must not be null')];
+      }
+      if (expr.value === undefined) {
+        return [createError('Type must not be undefined')];
+      }
+      if (expr.type === Syntax.Literal && expr.value != null && !args.has(expr.value.toString())) {
+        return [createError('Args must have value')];
+      }
+      return [];
+    }
   }
   if (isCategoryDecorator(decoratorName as Decorator)) {
-    return (expr: Expression): boolean =>
+    return (expr: Expression): DiagnosticError[] => {
       // this only verifies the usage, not correctness
       // (it will not check that the category is valid)
-      expr.type === Syntax.Literal && expr.value !== null && expr.value !== undefined;
+      if (expr.type !== Syntax.Literal) {
+        return [createError('Type must be Literal')];
+      }
+      if (expr.value === null) {
+        return [createError('Type must not be null')];
+      }
+      if (expr.value === undefined) {
+        return [createError('Type must not be undefined')];
+      }
```

Comment:
I think we can extract this in a helper checker `nonNullUndefinedLiteral` or smth.